{
  "comments": [
    {
      "unresolved": true,
      "key": {
        "uuid": "a1f329e9_825e695e",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 24
      },
      "lineNbr": 0,
      "author": {
        "id": 1359249
      },
      "writtenOn": "2021-06-21T21:10:48Z",
      "side": 1,
      "message": "@Shahbaz, results from last run -\nIn this latest run, I disabled getting Rasterization and ProvokingVertex (RendererVk.cpp):\n\n    // Query line rasterization features\n    if (ExtensionFound(VK_EXT_LINE_RASTERIZATION_EXTENSION_NAME, deviceExtensionNames))\n    {\n//        vk::AddToPNextChain(\u0026deviceFeatures, \u0026mLineRasterizationFeatures);\n    }\n\n    // Query provoking vertex features\n    if (ExtensionFound(VK_EXT_PROVOKING_VERTEX_EXTENSION_NAME, deviceExtensionNames))\n    {\n//        vk::AddToPNextChain(\u0026deviceFeatures, \u0026mProvokingVertexFeatures);\n    }\n\n\nThe Validation errors associated with them are no longer seen (features disabled):\n\nI   22.105s run_tests_on_device(9B121FFAZ008XY)  [ RUN      ] dEQP.KHR_GLES31/shaders_layout_location_sampler_cube_shadow\nI   22.105s run_tests_on_device(9B121FFAZ008XY)  KHR-GLES31.shaders.layout_location.sampler_cube_shadow\nI   22.105s run_tests_on_device(9B121FFAZ008XY)  contextVk-\u003egetFeatures().bresenhamLineRasterization.enabled 0\nI   22.105s run_tests_on_device(9B121FFAZ008XY)  contextVk-\u003egetFeatures().provokingVertex.enabled 0\nI   22.105s run_tests_on_device(9B121FFAZ008XY)  contextVk-\u003egetFeatures().depthClamping.enabled 0\nI   22.105s run_tests_on_device(9B121FFAZ008XY)  contextVk-\u003egetFeatures().supportsTransformFeedbackExtension.enabled 0\nI   22.105s run_tests_on_device(9B121FFAZ008XY)  Start multisampleState \nI   22.105s run_tests_on_device(9B121FFAZ008XY)  Start depthStencilState \nI   22.105s run_tests_on_device(9B121FFAZ008XY)  Start blendState \nI   22.105s run_tests_on_device(9B121FFAZ008XY)  Start dynamicState \nI   22.105s run_tests_on_device(9B121FFAZ008XY)  Start createInfo \nI   22.105s run_tests_on_device(9B121FFAZ008XY)  Calling initGraphics \nI   22.105s run_tests_on_device(9B121FFAZ008XY)  vkCreateGraphicsPipelines Result: 0\nI   22.105s run_tests_on_device(9B121FFAZ008XY)  returned from initGraphics \nI   22.105s run_tests_on_device(9B121FFAZ008XY)  [       OK ] dEQP.KHR_GLES31/shaders_layout_location_sampler_cube_shadow (24 ms)\n\n\nHowever new errors arise - for test (dEQP.KHR_GLES31/core_draw_indirect_basicindicesDataTypeunsigned_byte) that was NOT present in previous run: (and once again Validation says that some setting is set, but feature is not enabled.\n\nI   59.628s run_tests_on_device(9B121FFAZ008XY)  [ RUN      ] dEQP.KHR_GLES31/core_draw_indirect_basicindicesDataTypeunsigned_byte\nI   59.628s run_tests_on_device(9B121FFAZ008XY)  KHR-GLES31.core.draw_indirect.basic-indicesDataType-unsigned_byte\nI   59.628s run_tests_on_device(9B121FFAZ008XY)  contextVk-\u003egetFeatures().bresenhamLineRasterization.enabled 0\nI   59.628s run_tests_on_device(9B121FFAZ008XY)  contextVk-\u003egetFeatures().provokingVertex.enabled 0\nI   59.628s run_tests_on_device(9B121FFAZ008XY)  contextVk-\u003egetFeatures().depthClamping.enabled 0\nI   59.628s run_tests_on_device(9B121FFAZ008XY)  contextVk-\u003egetFeatures().supportsTransformFeedbackExtension.enabled 0\nI   59.628s run_tests_on_device(9B121FFAZ008XY)  Start multisampleState \nI   59.628s run_tests_on_device(9B121FFAZ008XY)  Start depthStencilState \nI   59.628s run_tests_on_device(9B121FFAZ008XY)  Start blendState \nI   59.628s run_tests_on_device(9B121FFAZ008XY)  Start dynamicState \nI   59.628s run_tests_on_device(9B121FFAZ008XY)  Start createInfo \nI   59.629s run_tests_on_device(9B121FFAZ008XY)  Calling initGraphics \nI   59.629s run_tests_on_device(9B121FFAZ008XY)  vkCreateGraphicsPipelines Result: 0\nI   59.629s run_tests_on_device(9B121FFAZ008XY)  returned from initGraphics \nI   59.629s run_tests_on_device(9B121FFAZ008XY)  ../../src/tests/deqp_support/angle_deqp_gtest.cpp:51: Failure\nI   59.629s run_tests_on_device(9B121FFAZ008XY)  Failed\nI   59.629s run_tests_on_device(9B121FFAZ008XY)  RendererVk.cpp:362 (DebugUtilsMessenger): [ VUID-vkCmdBindIndexBuffer-indexType-02765 ] Validation Error: [ VUID-vkCmdBindIndexBuffer-indexType-02765 ] Object 0: handle \u003d 0xb400006f6f6cbdb0, type \u003d VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID \u003d 0x93a57546 | vkCmdBindIndexBuffer() indexType is VK_INDEX_TYPE_UINT8_EXT but indexTypeUint8 feature is not enabled. The Vulkan spec states: If indexType is VK_INDEX_TYPE_UINT8_EXT, the indexTypeUint8 feature must be enabled (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkCmdBindIndexBuffer-indexType-02765)\nI   59.629s run_tests_on_device(9B121FFAZ008XY)                              Object: 0xb400006f6f6cbdb0 (type \u003d Command Buffer(6))\nI   59.629s run_tests_on_device(9B121FFAZ008XY)  \nI   59.629s run_tests_on_device(9B121FFAZ008XY)  [  FAILED  ] dEQP.KHR_GLES31/core_draw_indirect_basicindicesDataTypeunsigned_byte, where GetParam() \u003d 2088 (26 ms)\n\n\nThe fact that the error are resolved, seems to indicate a issue with Validation Layer. The fact that new tests have new Validation errors is perplexing.",
      "revId": "040ba8a471540e614f87e616b8019e99d26b3054",
      "serverId": "3ce6091f-6c88-37e8-8c75-72f92ae8dfba"
    }
  ]
}